packages/editor/src/components/browser/DataTypeBrowser.tsx

```diff
@@ -1,13 +1,225 @@
+  );
+
+  const [expanded, setExpanded] = useState<ExpandedState>(true);
+  const [globalFilter, setGlobalFilter] = useState(mainFilter);
+  const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
+
+  const table = useReactTable({
+    data: datatypeList,
+    columns: columns,
+    state: {
+      expanded,
+      globalFilter,
+      rowSelection
+    },
+    filterFromLeafRows: true,
+    enableRowSelection: true,
+    enableMultiRowSelection: false,
+    enableSubRowSelection: false,
+    onExpandedChange: setExpanded,
+    onGlobalFilterChange: setGlobalFilter,
+    onRowSelectionChange: setRowSelection,
+    getCoreRowModel: getCoreRowModel(),
+    getExpandedRowModel: getExpandedRowModel(),
+    getFilteredRowModel: getFilteredRowModel()
+  });
+
+  const addListGeneric = (value: JavaType, typeAsList: boolean) => {
+    if (typeAsList) {
+      return 'java.util.List<' + value.simpleName + '>';
+    } else {
+      return value.fullQualifiedName;
+    }
+  };
+
+  useEffect(() => {
+    if (Object.keys(rowSelection).length !== 1) {
+      setSelectedDataType({ fullQualifiedName: '', packageName: '', simpleName: '' });
+      props.onChange('');
+      setShowHelper(false);
+      return;
+    }
+    const selectedRow = table.getRowModel().rowsById[Object.keys(rowSelection)[0]];
+    setSelectedDataType(selectedRow.original);
+    setShowHelper(true);
+    if (selectedRow.original.simpleName === 'List') {
+      setListtypeSelected(true);
+    } else {
+      setListtypeSelected(false);
+    }
+    props.onChange(addListGeneric(selectedRow.original, typeAsList));
+  }, [props, props.onChange, rowSelection, table, typeAsList]);
+
+  return (
+    <>
+      <Table
+        search={{
+          value: globalFilter,
+          onChange: newFilterValue => {
+            setGlobalFilter(newFilterValue);
+            setMainFilter(newFilterValue);
+            setExpanded(true);
+          }
+        }}
+      >
+        <tbody>
+          {table.getRowModel().rows.map(row => (
+            <SelectRow key={row.id} row={row}>
+              {row.getVisibleCells().map(cell => (
+                <TableCell key={cell.id}>{flexRender(cell.column.columnDef.cell, cell.getContext())}</TableCell>
+              ))}
+            </SelectRow>
+          ))}
+        </tbody>
+      </Table>
+      {!listtypeSelected && (
+        <>
+          {(props.value.length !== 0 && selectedDataType) || mainFilter === 'List' ? (
+            <>
+              {mainFilter === 'List' && typeAsList && (
+                <MessageText
+                  message={{
+                    severity: 'WARNING',
+                    message: `The "Use Type as List" checkmark is still enabled, causing lists not to be displayed.`
+                  }}
+                />
+              )}
+              <Checkbox label='Use Type as List' value={typeAsList} onChange={() => setTypeAsList(!typeAsList)} />
+            </>
+          ) : null}
+        </>
+      )}

```

Comment:
also here, I would keep it simple.
I would just filter the list type from the possible types (maybe even in the backend!)